### ISSUE TITLE
ccminer crashs on help display

### ISSUE THREAD

**Issue by PapaChealio** [KlausT/ccminer#177]
I know you're battling with app crashes, but the latest 8.19 final release also now crashes just on display of help text:

ccminer.exe --help

...causes "ccminer has stopped working" error in Windows after displaying help text.

**Comment by KlausT** [comment]
i think I'm the only one where ccminer doesn't crash at all.
It's really wierd.

**Comment by PapaChealio** [comment]
The two places I get crashes consistently are CTRL-C to abort the miner in console and, in the latest version, via help display.

I'm using 4 x 1080Ti + 1 x 1070 with 390.65 drivers on Windows 10 x64.

**Comment by UselessGuru** [comment]
> i think I'm the only one where ccminer doesn't crash at all.

No, there is hope. It's working fine here (Windows 10, Nvidia Driver 390.65)

**Comment by sukoshi1507** [comment]
Well we are two. And I'm running your ccminer with Linux ;)

Le 25 janv. 2018 00:19, "KlausT" <notifications@github.com> a écrit :

i think I'm the only one where ccminer doesn't crash at all.
It's really wierd.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/KlausT/ccminer/issues/177#issuecomment-360307658>, or mute
the thread

**Comment by PapaChealio** [comment]
To clarify, if you press CTRL-C multiple times to exit on the console, it crashes. If you just press CTRL-C once and wait 5-10 seconds (i.e. "x threads exiting, Stratum disconnected...", it exits gracefully. Also, it appears to exit gracefully when prompted via API.

**Comment by KlausT** [comment]
Yeah, I have noticed this yesterday

**Comment by leup** [comment]
Same here. Just did a `ccminer --help`. Also, when launching `ccminer` with no arguments =>crash.
Tried `ccminer -a lyra2z` (yeah did not see that it was not supported) => crash.

I would think that there is an issue when trying to use no algo or an unknown algo.

![image](https://user-images.githubusercontent.com/431119/35471127-a3f424e2-0355-11e8-9fad-cb7cf6681971.png)

**Comment by cgarnier** [comment]
I got this issue too on a 1070 TI rig with driver 390.11 on linux.
ccminer branch cuda9.
When i run ccminer, it just saying some "about" text and stop. I cant see the help.
This is the output:
```
ccminer --help
ccminer 8.19-KlausT (64bit) for nVidia GPUs
Compiled with GCC 5.4 using Nvidia CUDA Toolkit 9.1

Based on pooler cpuminer 2.3.2 and the tpruvot@github fork
CUDA support by Christian Buchner, Christian H. and DJM34
Includes optimizations implemented by sp-hash, klaust, tpruvot and tsiv.
```
